docs/template-development.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "Template"
+weight: 50
+---
+
+# Minimum template requirements
+
+The following section talks about the bare minimum components your template should have in any given situation:
+
+### 1. `template` directory
+
+This is where you keep the files that determine the output that will be generated. In other words, the Generator processes all the files stored in this directory.
+
+```js
+import { File, Text } from "@asyncapi/generator-react-sdk";
+
+export default function({ asyncapi, params, originalAsyncAPI }) {
+return (
+    <File name="asyncapi.md">
+    <Text>This is a markdown file for my application.</Text>
+    <Text>App name is: **{ asyncapi.info().title() }**</Text>
+    </File>
+);
+}
+```
+
+The above example of `template/index.js` file shows as a result of the generation process. A user of such a minimal template will receive `asyncapi.md` file with some hardcoded information and some dynamic information extracted for the AsyncAPI file.
+
+Important to notice is that every template must depend on [`@asyncapi/generator-react-sdk`](https://github.com/asyncapi/generator-react-sdk) package containing a set of basic components to use in template files.
+
+### 2. `package.json` file
+
+Before the generation process runs, the Generator must install the template into its dependencies, and `package.json` is necessary to identify the template name.
+
+Following block shows an example `package.json` file and some select dependencies:
+
+```json
+{
+  "name": "myTemplate",
+  "generator": {
+    "renderer": "react"
+  },
+  "dependencies": {
+    "@asyncapi/generator-react-sdk": "^0.2.25"
+  }
+}
+```
+
+> Whenever you make a change to package.json make sure you perform an update with `npm install` to synchronize with package-lock.json and validate if the file is not broken after your changes.
+
```

Comment:
please add here first a section about template configuration, short section where you intro configuration concept and link to proper file for more (https://github.com/asyncapi/generator/blob/master/docs/configuration-file.md)